sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SynchronousInstrumentAccumulatorTest.java

```diff
@@ -27,12 +34,52 @@
   private final Aggregator<Long> aggregator =
       AggregatorFactory.lastValue()
           .create(Resource.empty(), InstrumentationLibraryInfo.create("test", "1.0"), DESCRIPTOR);
+  private final LabelsProcessor labelsProcessor =
+      LabelsProcessorFactory.noop()
+          .create(Resource.empty(), InstrumentationLibraryInfo.create("test", "1.0"), DESCRIPTOR);
+
+  @Test
+  void labelsProcessor_used() {
+    LabelsProcessor spyLabelsProcessor = Mockito.spy(this.labelsProcessor);
+    SynchronousInstrumentAccumulator<?> accumulator =
+        new SynchronousInstrumentAccumulator<>(
+            aggregator, new InstrumentProcessor<>(aggregator, testClock.now()), spyLabelsProcessor);
+    accumulator.bind(Labels.empty());
+    Mockito.verify(spyLabelsProcessor).onLabelsBound(Context.current(), Labels.empty());
+  }
+
+  @Test
+  void labelsProcessor_applied() {
+    final Labels labels = Labels.of("K", "V");
+    LabelsProcessor labelsProcessor =
+        new LabelsProcessor() {
+          @Override
+          public Labels onLabelsBound(Context ctx, Labels lbls) {
+            return lbls.toBuilder().put("modifiedK", "modifiedV").build();
+          }
+        };
+    LabelsProcessor spyLabelsProcessor = Mockito.spy(labelsProcessor);
+    SynchronousInstrumentAccumulator<?> accumulator =
+        new SynchronousInstrumentAccumulator<>(
+            aggregator, new InstrumentProcessor<>(aggregator, testClock.now()), spyLabelsProcessor);
+    AggregatorHandle<?> aggregatorHandle = accumulator.bind(labels);
+    aggregatorHandle.recordDouble(1);
+    List<MetricData> md = accumulator.collectAll(testClock.now());
+    md.stream()
+        .flatMap(m -> m.getLongGaugeData().getPoints().stream())
+        .forEach(
+            p ->
+                assertThat(new HashMap<>(p.getLabels().asMap()))
```

Comment:
just a note: once #2963 is merged, I don't think we'll need these extra HashMap wrappers to properly do the equality checks.

Reply:
Please fix now

Reply:
done

Reply:
Still here the HashMap